pkg/core/workflow_store.go

```diff
@@ -62,44 +62,85 @@ func (ws *WorkflowStore) AddWorkflow(ctx context.Context, w *workflow.Workflow)
 	return w, nil
 }
 
+// DeleteWorkflow deletes the workflow from the store
+func (ws *WorkflowStore) DeleteWorkflow(ctx context.Context, name string) error {
+	sw, found := ws.items[name]
+	if !found {
+		return nil
+	}
+	if len(sw.assignedCodesets) > 0 {
+		return fmt.Errorf("cannot delete workflow, there are codesets assigned to it")
+	}
+	delete(ws.items, name)
+	return nil
+}
+
 // GetAssignedCodesets returns a list of codesets assigned to the specified workflow
-func (ws *WorkflowStore) GetAssignedCodesets(ctx context.Context, workflowName string) []*domain.Codeset {
+func (ws *WorkflowStore) GetAssignedCodesets(ctx context.Context, workflowName string) []*domain.AssignedCodeset {
 	if _, exists := ws.items[workflowName]; exists {
 		return ws.items[workflowName].assignedCodesets
 	}
 	return nil
 }
 
 // GetAssignments returns a map of workflows and its assigned codesets
-func (ws *WorkflowStore) GetAssignments(ctx context.Context, workflowName *string) (result map[string][]*domain.Codeset) {
-	result = make(map[string][]*domain.Codeset, len(ws.items))
+func (ws *WorkflowStore) GetAssignments(ctx context.Context, workflowName *string) (result map[string][]*domain.AssignedCodeset) {
+	result = make(map[string][]*domain.AssignedCodeset, len(ws.items))
```

Comment:
Not sure how relevant this is, given that this an in-memory store not to be used in production: I wouldn't initialize this to the size of the workflow store. Better to let it grow as you add results to it then to return a huge map of which you only use one entry.